runtime/io.js

```diff
@@ -171,68 +216,64 @@ function caml_ml_open_descriptor_out_with_flags(fd, flags){
 }
 
 //Provides: caml_channel_descriptor
-//Requires: caml_ml_channels
 //Alias: win_filedescr_of_channel
 function caml_channel_descriptor(chanid){
-  var chan = caml_ml_channels[chanid];
+  var chan = chanid.get();
   return chan.fd;
 }
 
 //Provides: caml_ml_set_binary_mode
-//Requires: caml_ml_channels
 function caml_ml_set_binary_mode(chanid,mode){
-  var chan = caml_ml_channels[chanid];
+  var chan = chanid.get();
   chan.file.flags.text = !mode
   chan.file.flags.binary = mode
   return 0;
 }
 
 //Provides: caml_ml_is_binary_mode
-//Requires: caml_ml_channels
 //Version: >= 5.2
 function caml_ml_is_binary_mode(chanid) {
-  var chan = caml_ml_channels[chanid];
+  var chan = chanid.get();
   return chan.file.flags.binary 
 }
 
 //Input from in_channel
 
 //Provides: caml_ml_close_channel
-//Requires: caml_ml_flush, caml_ml_channels
+//Requires: caml_ml_flush
 //Requires: caml_sys_close
 function caml_ml_close_channel (chanid) {
-  var chan = caml_ml_channels[chanid];
+  var chan = chanid.get();
+  if (!chan.opened) { return 0; }
   chan.opened = false;
-  caml_sys_close(chan.fd)
+  caml_sys_close(chan.fd);
+  chanid.close();
```

Comment:
What happen if you close an "overridden" channel twice ?
Don't you end up closing both the override and the original channel ?